internal/handlers/pg/cmd_query.go

```diff
@@ -33,6 +33,8 @@ import (
 
 // CmdQuery implements HandlerInterface.
 func (h *Handler) CmdQuery(ctx context.Context, query *wire.OpQuery) (*wire.OpReply, error) {
+	common.CheckClientMetadata(ctx, query.Query)
```

Comment:
I think we should do it below, where we check for "ismaster", and don't ignore error